### PR TITLE
Remove BuildAndTest and RebuildAndTest target in favor of Test

### PR DESCRIPTION
**Pull request by ViktorHofer**
To get closer to the VSTest behavior this changes the Test target to
implicitly invoke the Build target unless /p:TestNoBuild is passed in.
(VSTest uses VSTestNoBuild which is controlled by the
dotnet test --no-build flag)

While this brings us onto a more common path this also simplifies the local dev flow:
- `dotnet msbuild /t:RebuildAndTest` or `dotnet msbuild /t:BuildAndTest` -> `dotnet build /t:Test`
- `dotnet msbuild /t:Test` -> `dotnet build /t:Test /p:TestNoBuild=true`

Running tests without building them is less common therefore I'm fine with the additional necessary parameter if the build should be skipped.

### CODE REVIEW

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -598,7 +598,7 @@ function MSBuild() {
 }
 
 #
-# Executes msbuild (or 'dotnet msbuild') with arguments passed to the function.
+# Executes msbuild (or 'dotnet build') with arguments passed to the function.
```
Comment (akoeplinger) [inline eng/common/tools.ps1]: changes in eng/common should probably be reverted and done in arcade

Comment (ViktorHofer) [reply eng/common/tools.ps1]: thanks

**Comment by hughbe** [comment]
Sorry if I’ve missed this somewhere but can we use `dotnet test` Here?

**Comment by ViktorHofer** [comment]
Unfortunately not yet. We have asked the VSTest team for the required features and are currently in design discussions. The main problem is routing to the correct dotnet host and shared framework.

**Comment by marek-safar** [comment]
> We have asked the VSTest team for the required features and are currently in design discussions

@ViktorHofer could you please forward me the discussion? The requirements should reflect the .net5 extensions for additional non-shared runtime used with `--vm`

**Comment by ViktorHofer** [comment]
What is the - - vm switch? A dotnet host option?

File: eng/testing/tests.targets [hunk 2]
```diff
@@ -139,9 +134,7 @@
   <Import Project="$(MSBuildThisFileDirectory)xunit\xunit.targets" Condition="'$(TestFramework)' == 'xunit'" />
 
   <!-- Main test targets -->
-  <Target Name="Test" DependsOnTargets="GenerateRunScript;RunTests" />
-  <Target Name="BuildAndTest" DependsOnTargets="Build;Test" />
-  <Target Name="RebuildAndTest" DependsOnTargets="Rebuild;Test" />
+  <Target Name="Test" DependsOnTargets="$(TestDependsOn)" />
```
Comment (Anipik) [inline eng/testing/tests.targets]: Do we need to  update the outerBuild "Test" Target ?

File: src/libraries/dir.traversal.targets [hunk 3]
```diff
@@ -46,11 +46,8 @@
   </PropertyGroup>
 
   <Target Name="Build" DependsOnTargets="$(TraversalBuildDependsOn)" />
-  <Target Name="BuildAndTest" DependsOnTargets="Build;Test" />
   <Target Name="Clean" DependsOnTargets="$(TraversalCleanDependsOn)" />
   <Target Name="Rebuild" DependsOnTargets="Clean;Build" />
-  <Target Name="RebuildAndTest" DependsOnTargets="Rebuild;Test" />
   <Target Name="Restore" />
-  <Target Name="Test" />
```
Comment (Anipik) [inline src/libraries/dir.traversal.targets]: Why are we removing the test target from here ?

File: src/libraries/tests.proj [hunk 4]
```diff
@@ -2,13 +2,14 @@
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.props" />
   <Import Project="$(RepositoryEngineeringDir)coverage.props" Condition="'$(EnableCoverageSupport)' == 'true'" />
 
+  <PropertyGroup>
+    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
```
Comment (Anipik) [inline src/libraries/tests.proj]: Why do we need to specify the TargetFramework here ?

**Review by Anipik — COMMENTED** [review]
just few questions.

File: eng/testing/tests.targets (hunk 2 above)
Comment (ViktorHofer) [reply eng/testing/tests.targets]: No

**Review by Anipik — APPROVED** [review]

File: src/libraries/dir.traversal.targets (hunk 3 above)
Comment (ViktorHofer) [reply src/libraries/dir.traversal.targets]: That target is dead code.

**Comment by ViktorHofer** [comment]
Failure is https://github.com/NuGet/Home/issues/9199

**Review by safern — COMMENTED** [review]
LGTM.

A comment, I don't see any mention of `/p:NoTestBuild=true` in the docs update.

**Pull request merged**